redfish-core/lib/assembly.hpp

```diff
@@ -0,0 +1,494 @@
+
+        nlohmann::json::object_t item;
+        item["@odata.type"] = "#Assembly.v1_3_0.AssemblyData";
+        item["@odata.id"] = dataID;
+        item["MemberId"] = std::to_string(assemblyIndex);
+
+        tempyArray.emplace_back(item);
+
+        tempyArray.at(assemblyIndex)["Name"] =
+            sdbusplus::message::object_path(assembly).filename();
+
+        dbus::utility::getDbusObject(
+            assembly, interfaces,
+            [asyncResp, assemblyIndex,
+             assembly](const boost::system::error_code ec,
+                       const dbus::utility::MapperGetObject& object) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("DBUS response error : {}", ec.value());
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            for (const auto& [serviceName, interfaceList] : object)
+            {
+                for (const auto& interface : interfaceList)
+                {
+                    if (interface ==
+                        "xyz.openbmc_project.Inventory.Decorator.Asset")
+                    {
+                        getAssemblyAsset(asyncResp, serviceName, assembly,
+                                         assemblyIndex);
+                    }
+                    else if (interface == "xyz.openbmc_project.Inventory."
+                                          "Decorator.LocationCode")
+                    {
+                        getAssemblyLocationCode(asyncResp, serviceName,
+                                                assembly, assemblyIndex);
+                    }
+                }
+            }
+        });
+
+        nlohmann::json::array_t assemblyArray =
+            asyncResp->res.jsonValue["Assemblies"];
+        asyncResp->res.jsonValue["Assemblies@odata.count"] =
+            assemblyArray.size();
+
+        assemblyIndex++;
+    }
+}
+
+/**
+ * @brief Api to check if the assemblies fetched from association Json is
+ * also implemented in the system. In case the interface for that assembly
+ * is not found update the list and fetch properties for only implemented
+ * assemblies.
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisPath - Chassis the assemblies are associated with.
+ * @param[in] assemblies - list of all the assemblies associated with the
+ * chassis.
+ * @return None.
+ */
+inline void
+    checkAssemblyInterface(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                           const std::string& chassisPath,
+                           const dbus::utility::MapperEndPoints& assemblies)
+{
+    constexpr std::array<std::string_view, 5> fruInterface = {
+        "xyz.openbmc_project.Inventory.Item.Vrm",
+        "xyz.openbmc_project.Inventory.Item.Tpm",
+        "xyz.openbmc_project.Inventory.Item.Panel",
+        "xyz.openbmc_project.Inventory.Item.Battery",
+        "xyz.openbmc_project.Inventory.Item.DiskBackplane"};
```

Comment:
In 1060, the interfaces for this `checkAssemblyInterface()` are more than this list.
https://github.com/ibm-openbmc/bmcweb/blob/fcd14caf229caca8fb529fe4d3b56da2dfcd939b/redfish-core/include/utils/chassis_utils.hpp#L67C1-L82C1

Should these also be added?

        "xyz.openbmc_project.Inventory.Item.Board",
        "xyz.openbmc_project.Inventory.Item.Connector",
        "xyz.openbmc_project.Inventory.Item.Drive",

Reply:
Understood. Thanks